soroban-spec/src/gen/json.rs

```diff
@@ -50,10 +51,12 @@ pub fn generate_from_wasm(wasm: &[u8]) -> Result<String, FromWasmError> {
 }
 
 pub fn generate(spec: &[ScSpecEntry]) -> String {
-    spec.iter()
-        .map(Entry::from)
-        .map(|e| serde_json::to_string_pretty(&e).expect("serialization of the spec entries should not have any failure cases as all keys are strings and the serialize implementations are derived"))
-        .collect()
+    format!("[{}]",
+        spec.iter()
+            .map(Entry::from)
+            .map(|e| serde_json::to_string_pretty(&e).expect("serialization of the spec entries should not have any failure cases as all keys are strings and the serialize implementations are derived"))
+            .join(",")
+    )
```

Comment:
The current output is a valid JSON stream where each entity is a JSON object. Many tools will support it, although for some it might be inconvenient. For example, you can pipe the stream to tools like `jq` to operate on it.

What situations are you seeing this as invalid?

Making it a JSON array seems reasonable way of increasing interop.

Reply:
I was trying to import the output in javascript and the JSON parse complains.